src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -102,64 +113,67 @@ public NuGetProject DefaultNuGetProject
                 throw new ArgumentNullException(nameof(projectSystemFactory));
             }
 
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _serviceProvider = serviceProvider;
             _projectSystemCache = projectSystemCache;
             _projectSystemFactory = projectSystemFactory;
+            _logger = logger;
+        }
 
-            _dte = ServiceLocator.GetInstance<DTE>();
-            _vsSolution = ServiceLocator.GetGlobalService<SVsSolution, IVsSolution>();
-            _vsMonitorSelection = ServiceLocator.GetGlobalService<SVsShellMonitorSelection, IVsMonitorSelection>();
+        public async Task InitializeAsync(IAsyncServiceProvider site)
+        {
```

Comment:
I designed it the way all `InitializeAsync` are called in `RestoreManagerPackage` which is loaded on solution new/open. So it shouldn't be missed as I considered different scenarios.